Sources/MapboxDirections/VisualInstructionComponent.swift

```diff
@@ -104,18 +109,48 @@ public extension VisualInstruction.Component {
             case svg
         }
         
-        /**
-         Initializes an image representation bearing the image at the given base URL.
-         */
-        public init(imageBaseURL: URL?) {
+        /// Initializes an image representation bearing the image at the given base URL.
+        /// - Parameters:
+        ///   - imageBaseURL: The image's base URL.
+        ///   - availableScales: The image’s available scale factors. If this argument is unspecified, the current screen’s native scale factor is used.
+        ///   - availableFormats: The available file formats of the image.
+        public init(imageBaseURL: URL?, availableScales: Set<CGFloat>? = nil, availableFormats: Set<Format>) {
             self.imageBaseURL = imageBaseURL
+            self.imageURL = nil
+            
+            let defaultScales: Set<CGFloat> = [ImageRepresentation.currentScale]
+            self.availableScales = availableScales ?? defaultScales
+            self.availableFormats = availableFormats
+        }
+        
+        /// Initializes an image representation bearing the image at the given URL.
+        /// - Parameters:
+        ///   - imageURL: The image URL.
+        ///   - availableScales: The image’s available scale factors. If this argument is unspecified, the current screen’s native scale factor is used.
+        ///   - availableFormats: The available file formats of the image.
+        public init(imageURL: URL?, availableScales: Set<CGFloat>? = nil, availableFormats: Set<Format>) {
+            self.imageURL = imageURL
+            self.imageBaseURL = nil
+            
+            let defaultScales: Set<CGFloat> = [ImageRepresentation.currentScale]
+            self.availableScales = availableScales ?? defaultScales
+            self.availableFormats = availableFormats
         }
         
         /**
          The URL whose path is the prefix of all the possible URLs returned by `imageURL(scale:format:)`.
          */
         public let imageBaseURL: URL?
         
+        /// The image URL.
+        public let imageURL: URL?
+        
+        /// The image's available scales.
+        public let availableScales: Set<CGFloat>
+        
+        /// The image's available formats.
+        public let availableFormats: Set<Format>
```

Comment:
These properties aren’t being encoded, decoded, and compared explicitly because they’re derived from the component type, which is already being encoded, decoded, and compared.